store/iavl/store.go

```diff
@@ -275,31 +277,36 @@ func (st *Store) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 			break
 		}
 
+		_, res.Value = tree.GetVersioned(key, res.Height)
 		if req.Prove {
-			value, proof, err := tree.GetVersionedWithProof(key, res.Height)
-			if err != nil {
-				res.Log = err.Error()
-				break
-			}
-			if proof == nil {
-				// Proof == nil implies that the store is empty.
-				if value != nil {
-					panic("unexpected value for an empty proof")
+			var commitmentProof *ics23.CommitmentProof
+			var err error
+			if res.Value != nil {
+				// Only support query proof from MutableTree for now
+				mtree, ok := tree.(*iavl.MutableTree)
+				if !ok {
+					return sdkerrors.QueryResult(sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "store must contain iavl.MutableTree to return proof"))
 				}
-			}
-			if value != nil {
 				// value was found
-				res.Value = value
-				res.Proof = &merkle.Proof{Ops: []merkle.ProofOp{iavl.NewValueOp(key, proof).ProofOp()}}
+				commitmentProof, err = ics23iavl.CreateMembershipProof(mtree, req.Data)
+				if err != nil {
+					panic(fmt.Sprintf("unexpected value for empty proof: %s", err.Error()))
```

Comment:
I don't understand why we both panic and return and error. It seems like we should always return an error.

Reply:
This should never happen, so I panic. Also believe this matches previous behavior